gcc.yaml

```diff
@@ -1,14 +1,17 @@
 package:
   name: gcc
   version: "15.2.0"
-  epoch: 3
+  epoch: 4
   description: "the GNU compiler collection"
   copyright:
     - license: GPL-3.0-or-later WITH GCC-exception-3.1
   resources:
     cpu: 16
     memory: 16Gi
   dependencies:
+    provides:
+      - gcc-${{vars.major-version}}=${{package.full-version}}
```

Comment:
Thanks! I feel like we should also include gcc-15/g++-15 executables if we provide this.

Reply:
After running into another problem today, I couldn't agree more. Will get a PR up in a moment